internal/metamorph/store/postgresql/postgres.go

```diff
@@ -525,19 +552,44 @@ func (p *PostgreSQL) UpdateMined(ctx context.Context, txsBlocks *blocktx_api.Tra
 		,t.retries
 		;
 `
-	rows, err := p.db.QueryContext(ctx, qBulkUpdate, metamorph_api.Status_MINED, p.now(), pq.Array(txHashes), pq.Array(blockHashes), pq.Array(blockHeights), pq.Array(merklePaths))
+	tx, err := p.db.Begin()
 	if err != nil {
-		return nil, fmt.Errorf("failed to execute transaction update query: %v", err)
+		panic(err)
 	}
 
-	return p.getStoreDataFromRows(rows)
+	_, err = tx.Exec("LOCK TABLE metamorph.transactions IN EXCLUSIVE MODE")
```

Comment:
We haven't seen a deadlock issue in `UpdateMined`. I would suggest to only do the change in `UpdateStatusBulk` for now and see what happens.

Reply:
I think it's just because it was less probably. It does the same thing - updates bunch of rows in the database. So it would be better to protect ourselves.

Reply:
OK. Let's see what is the performance penalty of locking the table for each call.